api/src/core/adapters/hal/getHalSoftware.ts

```diff
@@ -59,3 +57,29 @@ export async function fetchHalSoftwareById(halDocid: string): Promise<HalRawSoft
 
     return json.response.docs[0];
 }
+
+export async function fetchHalSoftwares(): Promise<Array<HalRawSoftware>> {
+    // Filter only software who have an swhidId to filter clean data on https://hal.science, TODO remove and set it as an option to be generic
+    const url = `https://api.archives-ouvertes.fr/search/?q=docType_s:SOFTWARE&rows=10000&fl=${halSoftwareFieldsToReturnAsString}&&fq=swhidId_s:["" TO *]`;
+
+    const res = await fetch(url).catch(() => undefined);
+
+    console.debug("Hal response status : ", res?.status);
+
+    if (res === undefined) {
+        throw new HalFetchError(undefined);
+    }
```

Comment:
tu peux faire ton throw dans le catch directement.

```suggestion
    const res = await fetch(url).catch((error) => {
            console.error(error);
            throw new HalFetchError(undefined);
    });
```